src/main.rs

```diff
@@ -51,26 +52,36 @@ fn main() {
 
 fn run() -> Result<(), MainError> {
     let cli = Cli::parse();
+    let cwd_and_manifest = || -> Result<_, MainError> {
+        let mut cwd = env::current_dir().expect("Couldn't find the current working directory");
+        if cli.workspace_root {
+            cwd = workspace::find_workspace_root(&cwd)?;
+        }
+        let manifest_path = cwd.join("package.json");
+        let manifest = read_package_manifest(&manifest_path)?;
+        Ok((cwd, manifest))
+    };
+    let print_and_run_script = |manifest: &NodeManifest, name: &str, command: &str, cwd: &Path| {
+        eprintln!(
+            "\n> {name}@{version} {cwd}",
+            name = &manifest.name,
+            version = &manifest.version,
+            cwd = dunce::canonicalize(cwd)
+                .unwrap_or_else(|_| cwd.to_path_buf())
+                .display(),
+        );
+        eprintln!("> {command}\n");
+        run_script(name, command, cwd)
+    };
     match cli.command {
+        cli::Command::PnpmCommands(c) => {
+            handle_passed_through(format!("{c}").as_str(), PassedThroughArgs::default())
```

Comment:
Normally, I would have said `c.to_string()` is preferable to `format!("{c}")`, but both still copy and allocate.

You should have derived `strum::AsRefStr` and called `c.as_ref()`.

Reply:
...if `handle_passed_through` is even necessary that is!

It turns out that all that `handle_passed_through` does is to prepend command name to the list of args and pass to pnpm. The command name and the list of args were both disassembled by `clap::Parser`, but now that it no longer necessary, `handle_passed_through` just becomes unnecessary extra work.